application/src/main/java/net/furizon/backend/feature/pretix/event/usecase/ReloadEventsUseCase.java

```diff
@@ -0,0 +1,118 @@
+package net.furizon.backend.feature.pretix.event.usecase;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.furizon.backend.feature.pretix.event.Event;
+import net.furizon.backend.feature.pretix.event.PretixEvent;
+import net.furizon.backend.feature.pretix.event.action.insert.InsertNewEventAction;
+import net.furizon.backend.feature.pretix.event.action.update.UpdateEventAction;
+import net.furizon.backend.feature.pretix.event.finder.EventFinder;
+import net.furizon.backend.feature.pretix.event.finder.pretix.PretixEventFinder;
+import net.furizon.backend.feature.pretix.organizer.PretixOrganizer;
+import net.furizon.backend.feature.pretix.organizer.finder.OrganizersFinder;
+import net.furizon.backend.infrastructure.pretix.PretixConfig;
+import net.furizon.backend.infrastructure.pretix.PretixPagingUtil;
+import net.furizon.backend.infrastructure.usecase.UseCase;
+import net.furizon.backend.infrastructure.usecase.UseCaseInput;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * This use case should get all events from Pretix
+ * Insert it to Database if not exist there (yet)
+ * Returns the current event object or null if non was found
+ */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class ReloadEventsUseCase implements UseCase<UseCaseInput, Optional<Event>> {
+    private final OrganizersFinder organizersFinder;
+    private final PretixEventFinder pretixEventFinder;
+    private final EventFinder eventFinder;
+
+    private final InsertNewEventAction insertEventAction;
+    private final UpdateEventAction updateEventAction;
+
+    private final PretixConfig pretixConfig;
+
+    @Transactional
+    @NotNull
+    @Override
+    public Optional<Event> executor(@NotNull UseCaseInput input) {
+        AtomicReference<Event> currentEvent = new AtomicReference<>(null);
+        PretixPagingUtil.fetchAll(
```

Comment:
maybe it's better to get all organizers first and after that we can iterate by it? wdyt? 
```java
PretixPagingUtil.combineAll(organizersFinder::getPagedOrganizers)
  .forEach(...);
 // or
 final var organizers = PretixPagingUtil.combineAll()
 for(final var organizer : organizers) {
   // and here we can remove AtomicReference<Event>, just use Event currentEvent = null; because this is not labda
   // TODO -> Things
 }
```